Add additonal Stm32g0 socs and nucleog0b1 board.

str4t0m opened the pull request:
- dts/arm/st: g0: remove unnecessary interrupt-names from adc, and uart
- dts/arm/st: add additional g0 device definitions
- soc/arm/st_stm32: Add additional stm32g0 socs
- drivers: gpio: add definition for stm32g0 porte
- drivers: clock_control: stm32g0: add support for g051, g0b1, g0b0 socs
- board: arm: add support for the nucleo_g0b1re board
- drivers: clock_control: stm32: rm Kconfig for APB2 prescaler on g0 series
- boards: stm32g0: add jlink and stm32cubeprogrammer runners.

Flashing of stm32g0b1 currently works with stm32cubeprogrammer and jlink.
Keil has already released a dsp for the new series, but some issues lead to pyocd flashing failure,
therefore i have added stm32g071 as target for pyocd for now.

str4t0m committed:
dts/arm/st: add additional g0 device defintions

Restructures dts files(value line definitions are no longer included by
others), and adds additional devices(g05x and g0bx).
All newly added timer pwm prescalers are set to 0.
lpuart1 is moved from stm32g0.dtsi to stm32g031.dtsi, as it is not
available on stm32g0x0 soc.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
soc/arm/st_stm32: Add additional stm32g0 socs

Add soc configuration for g05x and g0bx socs.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
board: arm: add support for the nucleo_g0b1re board

Support for the nucleo_g0b1re.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
drivers: gpio: add definition for stm32g0 porte

Advanced stm32g0 socs additionally have gpio port e.
This commit adds the missing definition for the port.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
drivers: clock_control: stm32g0: add support for g051, g0b1, g0b0 socs

Sets CLOCK_STM32_PLL_Q_DIVISOR for all g0x1 socs and g0b0.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
drivers: clock_control: stm32: rm Kconfig for APB2 prescaler on g0 serie

stm32g0 series does not have apb2 prescaler, reflect that in kconfig.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
boards: stm32g0: add jlink and stm32cubeprogrammer runners.

Add jlink and stm32cubeprogrammer runners to nucleo_g071rb and
stm32g0316_disco boards.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

erwango commented on dts/arm/st/g0/stm32g070X6.dtsi:
```diff
@@ -0,0 +1,22 @@
+/*
```
I haven't seen any reference to this one (only STM32G070CB STM32G070KB STM32070RB)
Do you have any link ?

str4t0m replied:
I have accidentally duplicated that from 71x6.

erwango commented on dts/arm/st/g0/stm32g070X8.dtsi:
```diff
@@ -0,0 +1,22 @@
+/*
```
Same here

str4t0m replied:
removed

erwango commented on dts/arm/st/g0/stm32g071X6.dtsi:
```diff
@@ -0,0 +1,22 @@
+/*
```
This one is not known neither

str4t0m replied:
It is listed in the Keil pack, but that could be an error. Should I remove it?

str4t0m replied:
I have removed it now as I only found this in the keil dfp and not on the ST website.

erwango commented on dts/arm/st/g0/stm32g0b0.dtsi:
```diff
@@ -0,0 +1,85 @@
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00008000>;
+			interrupts = <25 1>;
+			status = "disabled";
+			interrupt-names = "combined";
```
We're not using interrupt names on spi nodes

erwango commented on dts/arm/st/g0/stm32g0b1.dtsi:
```diff
@@ -0,0 +1,95 @@
+			reg = <0x40005000 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000100>;
+			interrupts = <29 2>;
+			interrupt-names = "combined";
```
Please remove interrupt-names

erwango commented on dts/arm/st/g0/stm32g0b1.dtsi:
```diff
@@ -0,0 +1,95 @@
+			compatible = "st,stm32-usart", "st,stm32-uart";
+			reg = <0x40013c00 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000200>;
+			interrupt-names = "combined";
```
Please remove interrupt-names

erwango commented on dts/arm/st/g0/stm32g0b1.dtsi:
```diff
@@ -0,0 +1,95 @@
+			reg = <0x40008400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000080>;
+			interrupts = <28 1>;
+			interrupt-names = "combined";
```
Please remove interrupt-names

erwango commented on dts/arm/st/g0/stm32g0b1.dtsi:
```diff
@@ -0,0 +1,95 @@
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00008000>;
+			interrupts = <25 1>;
+			status = "disabled";
+			interrupt-names = "combined";
```
Please remove interrupt-names

erwango commented on dts/arm/st/g0/stm32g0b1Xb.dtsi:
```diff
@@ -0,0 +1,22 @@
+
+/ {
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(144)>;
```
36

str4t0m replied:
The 144k should be right.

erwango commented:
@str4t0m Thanks for the work! Some changes required

str4t0m committed:
dts/arm/st: g0: remove superfluous interrupt-names from adc, and uart

"interrupt-names: combined" propterty is only used for stm32 i2c driver,
remove all other existing occurences for stm32g0(adc, lpuart, usart).

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m commented on dts/arm/st/g0/stm32g0.dtsi:
```diff
@@ -167,28 +157,101 @@
 			reg = <0x40004400 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00020000>;
 			interrupts = <28 0>;
+			interrupt-names = "combined";
```
is now removed.

str4t0m commented on dts/arm/st/g0/stm32g0.dtsi:
```diff
@@ -167,28 +157,101 @@
 			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000002>;
 			interrupts = <16 0>;
-			interrupt-names = "global";
+			interrupt-names = "combined";
```
I have left it "global", as there is currently no benefit in changing it.

str4t0m commented on dts/arm/st/g0/stm32g0.dtsi:
```diff
@@ -167,28 +157,101 @@
 				compatible = "st,stm32-pwm";
 				status = "disabled";
-				st,prescaler = <10000>;
+				st,prescaler = <0>;
```
This has now been moved into pr #32646

str4t0m commented on dts/arm/st/g0/stm32g031.dtsi:
```diff
@@ -1,7 +1,40 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <st/g0/stm32g0.dtsi>
```
done

str4t0m commented:
I have seen that the STM32_open_pin_data repo has been updated today (stm32cubemx 6.2.0 release).
- This is the first one that includes the definitions of the g05x socs.
- And for g0b1 it now includes the 128k flash variant:
  Additionally to the mcu/STM32G0B1R(C-E)Tx.xml they have now added a mcu/STM32G0B1R(B-C-E)Tx.xml  file...

Will we need to wait until the the pinctrl DTS files of the new release are merged in the hal before this pr can be merged, or could this pr be merged before?

str4t0m commented:
I addressed all changes and rebased, @erwango could you have another look?
